core/src/main/java/org/jruby/RubyThread.java

```diff
@@ -1029,6 +1029,15 @@ private Map<IRubyObject, IRubyObject> getThreadLocals0() {
         return locals;
     }
 
+    /**
+     * Clear the local variable storage for this thread.
+     * Meant for Java consumers when reusing threads (e.g. dur thread pooling).
+     */
+    public synchronized void clearLocalVariables() {
```

Comment:
I have a problem with this name 😀

Perhaps something like `clearConcurrentLocals` or some better name that indicates these are variables local to one of the two concurrency constructs?

Reply:
`clearVariableStores`?

Reply:
I am fine with whatever name you choose from the two, originally started with simply `clearLocals`.

I also can be convinced that having 2 `clearFiberLocals` and `clearThreadLocals` would be better...

Reply:
> I also can be convinced that having 2 clearFiberLocals and clearThreadLocals would be better...

I do like that idea. There are bound to be use cases for clearing one and not the other, e.g. Fiber-heavy framework wants to clear just Fiber locals after processing some chunk of work.

Reply:
updated, please let me know what you think...

also added a convenience static `RubyThread.current(runtime)` so it's easier to retrieve the current Ruby thread on the Java side.

Reply:
LGTM!